pkg/scalers/activemq_scaler.go

```diff
@@ -0,0 +1,282 @@
+}
+
+const (
+	defaultTargetQueueSize         = 10
+	defaultActiveMQRestAPITemplate = "http://{{.ManagementEndpoint}}/api/jolokia/read/org.apache.activemq:type=Broker,brokerName={{.BrokerName}},destinationType=Queue,destinationName={{.DestinationName}}/QueueSize"
+)
+
+var activeMQLog = logf.Log.WithName("activeMQ_scaler")
+
+// NewActiveMQScaler creates a new activeMQ Scaler
+func NewActiveMQScaler(config *ScalerConfig) (Scaler, error) {
+	meta, err := parseActiveMQMetadata(config)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing ActiveMQ metadata: %s", err)
+	}
+	httpClient := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout, false)
+
+	return &activeMQScaler{
+		metadata:   meta,
+		httpClient: httpClient,
+	}, nil
+}
+
+func parseActiveMQMetadata(config *ScalerConfig) (*activeMQMetadata, error) {
+	meta := activeMQMetadata{}
+
+	if val, ok := config.TriggerMetadata["restAPITemplate"]; ok && val != "" {
+		meta.restAPITemplate = config.TriggerMetadata["restAPITemplate"]
+		var err error
+		if meta, err = getRestAPIParameters(meta); err != nil {
+			return nil, fmt.Errorf("can't parse restAPITemplate : %s ", err)
+		}
+	} else {
+		meta.restAPITemplate = defaultActiveMQRestAPITemplate
+		if config.TriggerMetadata["managementEndpoint"] == "" {
+			return nil, errors.New("no management endpoint given")
+		}
+		meta.managementEndpoint = config.TriggerMetadata["managementEndpoint"]
+
+		if config.TriggerMetadata["destinationName"] == "" {
+			return nil, errors.New("no destination name given")
+		}
+		meta.destinationName = config.TriggerMetadata["destinationName"]
+
+		if config.TriggerMetadata["brokerName"] == "" {
+			return nil, errors.New("no broker name given")
+		}
+		meta.brokerName = config.TriggerMetadata["brokerName"]
+	}
+
+	if val, ok := config.TriggerMetadata["targetQueueSize"]; ok {
+		queueSize, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("invalid targetQueueSize - must be an integer")
+		}
+
+		meta.targetQueueSize = queueSize
+	} else {
+		meta.targetQueueSize = defaultTargetQueueSize
+	}
+
+	if val, ok := config.AuthParams["username"]; ok && val != "" {
+		meta.username = val
+	} else if val, ok := config.TriggerMetadata["username"]; ok && val != "" {
+		username := val
+
+		if val, ok := config.ResolvedEnv[username]; ok && val != "" {
+			meta.username = val
+		} else {
+			meta.username = username
+		}
+	}
+
+	if meta.username == "" {
+		return nil, fmt.Errorf("username cannot be empty")
+	}
+
+	if val, ok := config.AuthParams["password"]; ok && val != "" {
+		meta.password = val
+	} else if val, ok := config.TriggerMetadata["password"]; ok && val != "" {
+		password := val
+
+		if val, ok := config.ResolvedEnv[password]; ok && val != "" {
+			meta.password = val
+		} else {
+			meta.password = password
+		}
+	}
+
+	if meta.password == "" {
+		return nil, fmt.Errorf("password cannot be empty")
+	}
+
+	if val, ok := config.TriggerMetadata["metricName"]; ok {
```

Comment:
I wonder whether we should add this option to specify the `metricName`, since now we the safe generated one.

Reply:
you are right @zroubalik , maybe if the metricName is not used internally for other stuffs, we should avoid the usage of it in favor of easy use